Feat/splash screen + homeScreen + configScreen

Driemtax opened the pull request:
I added basic Views for showing:
- Splash Screen
- Home Screen
- Config Screen

I modified the WindowController so it orchestrates all views.

The Splash Screen shows the logo of BioCollapse and a fake progress Bar that takes some short time to fill. When the bar is filled the HomeScreen is shown, wich right now only conatins a singel Button to start the simulation. When pressed it shows the ConfigScreen wich is a tabbedPane. It contains 3 Tabs for Virus Parameters, Population Parameters and activated meassures. There is no saving Function for the Parameters nor a function to preload the the virus Parameters from external sources. This will be implemented in a seperate feature.

Driemtax committed:
feat: Added SplashScreen, no image yet

Driemtax committed:
feat: SplashScreen showing image now and HomeScreen is show aswell

Driemtax committed:
feat: showing ConfigScreen with tabbedPane, no saving of config yet

N3moAhead commented on src/com/biocollapse/view/SplashScreen.java:
```diff
@@ -0,0 +1,101 @@
+        panel.setLayout(new BorderLayout());
+
+        String imageUrl = System.getProperty("user.dir") + "/images/logo.png";
+        System.out.println("Bild: " + imageUrl);
```
Remove debug statements

N3moAhead commented on src/com/biocollapse/view/SplashScreen.java:
Looks very good to me, Im not sure what our Swing expert says.
Would be great if you could also have a look at it @joewilliams007 .

It would just be great if you could change german comments to english comments so everything is in english ;)

N3moAhead commented on src/com/biocollapse/controller/WindowController.java:
```diff
@@ -1,5 +1,41 @@
+    private ConfigScreen configScreen;
+    
+    public WindowController(){
+        frame = new JFrame("Biocollapse");
```
Biocollapse => BioCollapse

N3moAhead commented on src/com/biocollapse/view/ConfigScreen.java:
I spotted some german comments would be great if you could translate them to english :)

N3moAhead commented on src/com/biocollapse/view/HomeScreen.java:
I also spotted some wicked german comments in here 😈

N3moAhead reviewed (changes requested):
The Code to me looks very good! Thanks for the good work! 🙏🏻 

I would like @joewilliams007  to also look over it because im a stranger to Swing.

I found some german comments that I would like you to translate. But nothing codewise 👍🏻

Driemtax committed:
Merge branch 'main' into feat/configScreen

Driemtax committed:
feat: Added labels to config sliders and Measures to Config

joewilliams007 reviewed (commented):
Hi. Looks good so far! Some things to note.

1. Usually just extending a JPanel is cleaner than having a class maintaining one

2. In that regard, i would also rather call the panels. "ConfigPanel".

3. You are using the sliders and labels quite alot. Depending on the amount of UI elements it makes more sense to provide some sort of class or method which takes care of the creation of settings elements. And then creates these from a list of settings. That way you dont have hundred of lines creating ui elements. It doesnt matter too much in our case as maintenance wont be a thing that much.

5. The Strings of configScreen options shouldnt be hardcoded into the JLabels. They could be in a list of objects which hold maybe name, id, value and could also take care of a reference of their JSlider/Other Components.

6. Some comments are helpful but comments like e.g. tabbedPane.addTab(..) doesnt need a comment which says adding tab to tabbedPane

N3moAhead committed:
fix(MapPosition): Removed useless import

N3moAhead committed:
feat(Config): Added a global config provider

N3moAhead committed:
fix(Human): Replaced the local config through the GlobalConfig

N3moAhead committed:
fix: Removing unused imports and variables

Driemtax commented:
Thanks for all the feedback, that is very helpful as i am not experienced at all in frontend dev!

I made this PR because i wanted to continue on a new branch, but didnt think about the state of the code i wanted to merge. Sorry for that, i didnt think of anything like remove those comments.. 

I would suggest to merge this branch since i am already working on a new Branch to finish the config screen. In the new branch i would implement all your feedback so we have a clean code base to merge into main. Is that okay with you both? @N3moAhead @joewilliams007 

Maybe @joewilliams007 and i can have a quick chat on monday about JPanels, because i never worked with that before and im not completly sure what you mean with extending a JPanel. I would like learn about that!

Thanks for taking the time to review this PR!

N3moAhead commented:
@Driemtax, why don't we merge your new branch into this one and carry on from here? 👍🏻

N3moAhead committed:
Merge pull request #6 from N3moAhead/fix/using_global_config

Fix/using_global_config

Driemtax committed:
feat: Resolving comments on PR

Driemtax commented:
I think i resolved every comment. I changed:

1. Now extending JPanel and JFrame classes
2. Renamed the Panels
3. Implemented a method to create the labled sliders, because that was the most code. I think for this project this is enough, right?
4. This is not resolved yet, as i am not sure what exactly you mean. Maybe we should have a quick chat about this. The purpose i hardcoded JLabel strings is that it was the easiest and quickest way for me regarding the time for the project.
5. I deleted and translated comments

I also added all the config settings to the global config. I also resolved a bug where the title bar of the application wasn't shown because the JFrame was too big.

joewilliams007 reviewed (approved):
Heyy. This looks alot better now!

Regarding point 4, we can have a small chat about maybe.

Heck, i know a setup dialog (Tabbed Pane) which is 11.000 lines of UI by heart. And it has just as many JPanel extending class imports holding that many lines of code themselves.

It is a pain to maintain but dont get me started on it.. 😆 In a general use case, keeping things dynamic isnt more time consuming.

But for this project and as you mentioned the time, this is totally okay aswell.

Just one more thing, in home panel and splash panel, there are some JComponent variables that are private in class scope that dont have to be. Also, it is cleaner to skip using a variable (if you dont further need it) and just add the component right away.

E.g. Jlabel myLabel = new JLabel("hi")
this.add(myLabel)
Better to: this.add(new Jlabel("hi")); if the variable isnt needed

This is obviously nitpicking, and your code looks good to merge now!

joewilliams007 commented:
Woops, i think i havent set "seen" on all files on my review. If this blocks you from merging, feel free to request a review again, so i can approve all changes.

joewilliams007 reviewed (approved):
Approve according to my former review.
